adaptive_navigation/example/lib/custom_scaffold.dart

```diff
@@ -63,7 +63,7 @@ class _CustomScaffoldDemoState extends State<CustomScaffoldDemo> {
             value: _fabInRail,
             onChanged: (value) {
               setState(() {
-                _fabInRail = value;
+                _fabInRail = value!;
```

Comment:
can you help me understand what the ! does afterwards here?

Reply:
Due to current typing restrictions (see https://github.com/flutter/flutter/pull/67790#issuecomment-707920886), `Switch.onChanged` is a `void Function(bool?)?`, which means it must be a function that accepts a nullable boolean. Ideally it should be `void Function(bool)?`; in that case we wouldn't need the `!`. I think someone is working on that PR in the framework.